src/v/kafka/server/request_context.h

```diff
@@ -323,6 +324,62 @@ class request_context {
         return true;
     }
 
+    bool audit_authn_failure(const ss::sstring& reason) {
+        return audit_authn_failure(reason, "");
+    }
+
+    bool audit_authn_failure(
+      const ss::sstring& reason, const ss::sstring& auth_protocol) {
+        return audit_authn_failure(
+          reason,
+          auth_protocol,
+          {.type_id = security::audit::user::type::unknown});
+    }
+
+    bool audit_authn_failure(
+      const ss::sstring& reason,
+      const ss::sstring& auth_protocol,
+      security::audit::user user) {
+        return audit(security::audit::make_authentication_failure_event(
+          auth_protocol,
+          connection()->local_address(),
+          connection()->server().name(),
+          connection()->client_host(),
+          connection()->client_port(),
+          header().client_id,
+          connection()->tls_enabled()
+            ? security::audit::authentication::used_cleartext::no
+            : security::audit::authentication::used_cleartext::yes,
+          reason,
+          std::move(user)));
+    }
+
+    bool audit_authn_success(
+      const ss::sstring& auth_protocol, security::audit::user user) {
+        return audit(security::audit::make_authentication_event(
+          auth_protocol,
+          connection()->local_address(),
+          connection()->server().name(),
+          connection()->client_host(),
+          connection()->client_port(),
+          header().client_id,
+          connection()->tls_enabled()
+            ? security::audit::authentication::used_cleartext::no
+            : security::audit::authentication::used_cleartext::yes,
+          std::move(user)));
```

Comment:
Extension to the previous question: it seems like we could reduce some of the code change by passing the connection context

Reply:
I feel liek this could cause weird circular dependencies between kafka server and the audit system where kafka server depends on the audit library for its headers (utility functions for creating messages) and the audit library would depend on kafka server for its headers (connection_context)